### ISSUE TITLE
Support of  Shared Subscriptions

### ISSUE THREAD

**Issue by MrSpark2591** [mcollina/aedes#228]
I was reading article and found out [this article](https://www.hivemq.com/blog/mqtt-client-load-balancing-with-shared-subscriptions/)
Can it be done in this package ?
Or is there any other way we can achieve such scenario?

**Comment by mcollina** [comment]
Not at this point, you’ll have to code/provide that yourself.

**Comment by MrSpark2591** [comment]
I found out that we will need multiple changes for the support.

1) Multiple client session. currently only one client is allowed to connect to broker with that client id.
2) Mqtt.js support of v5 currently [under development](https://github.com/mqttjs/MQTT.js/issues/651).

I wil try if i can get any work around for this. let me know if i am missing anything here.

**Comment by mcollina** [comment]
The work on aedes is not in progress at all.

**Comment by MrSpark2591** [comment]
Is it due to support of mqtt or you need more maintainer to get it going? if anyways I can help let me know.

**Comment by mcollina** [comment]
I have no time to implement this at the moment.

**Comment by abhargav7** [comment]
any work around you guys found on it? That would be helpful Thanks!!

**Comment by robertsLando** [comment]
@gnought @mcollina This seems not that hard to implement, it's also part of mqtt5 but could be implemented without having mqtt5 support.

**Comment by abhargav7** [comment]
Thanks great to hear its quick. Its a powerful feature to have and solves lots of complex problems.

**Comment by robertsLando** [comment]
@abhargav7 I don't think it will be quick, it may require also an update of aedes-persistence

**Comment by abhargav7** [comment]
Sure, Thanks, when do you think you can contribute this for the betterment of aedes broker?

**Comment by robertsLando** [comment]
I have firstly to speak about this with others maintainers

**Comment by abhargav7** [comment]
Thanks, please!

**Comment by gnought** [comment]
I think we need a custom mqemitter.

**Comment by robertsLando** [comment]
Nope, we just need to create a sharedGroup client that keeps track of all clients subscribed to the group and sends the updates to only one client following round robin algorithm 

Daniel
